wdl/workflows/references/make_references.wdl

```diff
@@ -0,0 +1,128 @@
+    String genome_name = basename(reference_fasta, ".fa")
+    
+    if (!defined(bwa_index)) {
+        call bwa.build_bwa_index {
+            input:
+                fasta=reference_fasta,
+                resources=compute.runtimes[machine_size_bwa],
+        }
+    }
+
+    BwaIndex bwa_index_output = select_first([bwa_index, build_bwa_index.bwa_index])
+
+    call fai.build_fasta_index {
+        input:
+            fasta=reference_fasta,
+            resources=compute.runtimes[machine_size_fai],
+    }
+
+    if (!defined(bowtie_index)) {
+        call bowtie.build_bowtie_index {
+            input:
+                fasta=reference_fasta,
+                resources=compute.runtimes[machine_size_bowtie]
+        }
+    }
+
+    BowtieIndex bowtie_index_output = select_first([bowtie_index, build_bowtie_index.bowtie_index])
+
+    if (!defined(mappable_regions)) {
+        call mappable.build_mappable_only_bed {
+            input:
+                bowtie_index=bowtie_index_output,
+                kmer_length=mappability_kmer_length,
+                reference_genome=reference_fasta,
+                resources=compute.runtimes[machine_size_mappable],
+        }
+    }
+
+    File mappable_regions_output = select_first([mappable_regions, build_mappable_only_bed.mappable_regions])
+    
+    if (!defined(chrom_buckets)) {
+        call chr_buckets.get_chrombuckets {
+            input:
+                fai=select_first([build_fasta_index.indexed_fasta.fai]),
+                genome_name=genome_name,
+                resources=compute.runtimes[machine_size_chr_buckets],
+        }
+    }
+
+    File chrom_buckets_output = select_first([chrom_buckets, get_chrombuckets.chrombuckets])
+
+    if (!defined(chrom_sizes)) {
+        call chr_sizes.get_chrom_sizes {
+            input:
+                fai=select_first([build_fasta_index.indexed_fasta.fai]),
+                resources=compute.runtimes[machine_size_chr_sizes],
+        }
+    }
+
+    File chrom_sizes_output = select_first([chrom_sizes, get_chrom_sizes.chrom_sizes])
+
+    if (!defined(chrom_info)) {
+        call chr_info.get_chrom_info {
+            input:
+                chrom_sizes=chrom_sizes_output,
+                resources=compute.runtimes[machine_size_chr_info],
+        }
+    }
+
+    File chrom_info_output = select_first([chrom_info, get_chrom_info.chrom_info])
+
+    if (!defined(center_sites)) {
+        call center.get_center_sites {
+            input:
+                chrom_sizes=chrom_sizes_output,
+                mappable_regions=mappable_regions_output,
+                resources=compute.runtimes[machine_size_center],
+        }
+    }
+
+    File center_sites_output = select_first([center_sites, get_center_sites.center_sites_starch])
+
+    output {
+        BowtieIndex bowtie_index_out = bowtie_index_output
+        BwaIndex bwa_index_out = bwa_index_output
+        File center_sites_out = center_sites_output
```

Comment:
Should gather these in Hotspot1/2 references. Either https://github.com/ENCODE-DCC/dnase-seq-pipeline/blob/30d8d7b626e5b709931edd651c24bdd8a0ab987c/wdl/structs/dnase.wdl#L22 or its parts: https://github.com/ENCODE-DCC/dnase-seq-pipeline/blob/30d8d7b626e5b709931edd651c24bdd8a0ab987c/wdl/structs/hotspot2.wdl#L4 
And https://github.com/ENCODE-DCC/dnase-seq-pipeline/blob/30d8d7b626e5b709931edd651c24bdd8a0ab987c/wdl/structs/hotspot1.wdl#L4

Reply:
We also need something to sit next to dnase.wdl in base directory. Could probably move wdl/workflows/references/make_references.wdl to references.wdl (no make) in base directory.

Reply:
Opted for HS1, HS2 and IndexedFasta